doc/docs.md

```diff
@@ -6040,66 +6040,230 @@ To improve safety and maintainability, operator overloading is limited.
 - Arguments cannot be changed inside overloads.
 - Calling other functions inside operator functions is not allowed (**planned**).
 
-## Performance tuning
+## Tuning
 
-The generated C code is usually fast enough, when you compile your code
-with `-prod`. There are some situations though, where you may want to give
-additional hints to the compiler, so that it can further optimize some
-blocks of code.
+When compiled with `-prod`, V's generated C code usually performs well. However, in specialized
+scenarios, additional compiler flags and attributes can further optimize the executable for
+performance, memory usage, or size.
 
-> **Note**
-> These are *rarely* needed, and should not be used, unless you
-> *profile your code*, and then see that there are significant benefits for them.
-> To cite gcc's documentation: "programmers are notoriously bad at predicting
-> how their programs actually perform".
+> [!WARNING]
+> Use these options only after profiling your code and confirming their benefits.
+> Predicting performance without empirical data can be misleading.
```

Comment:
I prefer the old section, since it was both prescriptive, and descriptive.
The quote "programmers are notoriously bad at predicting how their programs actually perform" also has a more nuanced meaning than the bland statement "Predicting performance without empirical data can be misleading." .